### ISSUE TITLE
✈️ Run Elementor JavaScript After Ajax Request (Migrated)

### ISSUE THREAD

**Issue by dcooney** [elementor/elementor#12125]
I am loading Elementor posts with Ajax (infinite scroll) and various modules (Slider, Social Share, Posts Widget w/Masonry etc.) do not work after being loaded via Ajax.

This makes sense as each module requires JavaScript to initiate the functionality.

I've inspected all the frontend Elementor JS files but I'm unable to understand how (if possible) to trigger the required JS on specific modules.

For example, after a post is loaded via Ajax I need to call a function to start the Social Share module as the links are added via JavaScript, if I don't the links do not work. I have a callback in place to run the JavaScript, I just need to know if this is even possible with how Elementor JavaScript is compiled.

TL;DR - Is there a method to initiate an Elementor module after an Ajax request?

```
window.my_callback = function() {
   // Start Elementor modules
};
```

**Comment by dcooney** [comment]
I've tried WP Support forums, Elementor Pro support, the Elementor FB group and not a single person can answer this.
Is it at all possible to initiate Elementor modules JS after being loaded with Ajax?

**Comment by 27concepts** [comment]
Not sure I understand what you're trying to do. Where are those Elementor "modules" loaded from? Are they part of the post?

I think what you're trying to describe is maybe a template. If you need to reuse the social share in multiple places, save it as a template. That will give you a template it and a shortcode. 

Then, whenever you need to use it, you can make an Ajax call to a Wordpress function that simply returns the shortcode of the template.

**Comment by dcooney** [comment]
@27concepts I'm loading Elementor single posts via Ajax using Infinite Scroll.

Under normal conditions when a post is loaded, JavaScript is run on the initial page load to initiate the modules like Social Share, Carousel, Masonry (Posts Widget) etc.

However, when a post is loaded with Ajax, the required JavaScript is not run (this makes sense) and this means functionality is not working in the newly loaded content. 

I'm simply asking if the Elementor frontend JavaScript has methods/function I call to start the modules.

Hope that makes sense.

**Comment by 27concepts** [comment]
Maybe your approach here should be different. Like I said, you should look into loading a template instead of a single post. You build out your single post layout with Elementor, complete with whatever widgets you need, you save it as a template and then you load that (shortcode) template via an Ajax call. 

The shortcode will load whatever javascript and anything that template needs.

If this doesn't make sense, maybe someone else has other ideas.

**Comment by SuperstrongBE** [comment]
Of course it make sense ! 
Didi you find a way to implement that ?

**Comment by dcooney** [comment]
@SuperstrongBE I was able to solve the Masonry issue by just ripping apart the ElementorProFrontend JS.
https://examples.connekthq.com/elementor/elementor-posts-widget-masonry/

Not ideal as I had to recreate the entire custom Masonry function within my callback.

I have not been able to solve Share or Carousel

**Comment by SuperstrongBE** [comment]
Please comment/react on https://github.com/elementor/elementor/issues/11493 to make developer move on this

**Comment by loganvenderlic** [comment]
I'm looking for the almost the same thing! I'm pulling in other pages using admin-ajax calls and it doesn't bring in any of the javascript so none of the Elementor widgets work.

I've also tried just using jQuery's ajax and load functions. I actually got it to work by literally pulling in the entire page using those functions, but that doesn't feel like a great idea, because it pulls in duplicates of all of the styles and scripts.

I was hoping there would be a more elegant solution.

**Comment by ofmarconi** [comment]
@loganvenderlic Any progress?

**Comment by nicholaszein** [comment]
Hello.

I'm moving this issue to the [🌐 Developers Community](https://elemn.to/dev-community) Discussion Category as it is not really a bug. What you are trying to achieve is not supported by Elementor.

Moreover, custom code is out of scope for this repository.

Feel free to continue the conversation there.

Kind regards
